LIKA-412: update organization edit form

bertvi opened the pull request:
# Description
Update the ui for adding a new organization and editing an existing organization according to suomi.fi designs.

## Jira ticket reference: [LIKA-412](https://jira.dvv.fi/browse/LIKA-412)

## What has changed:
* Order of input fields
* Labels of input fields
* Placeholder values
* Submit/cancel button design and labels
* Make apicatalog_scheming translatable
* New translation strings and respective translations

## Checklist  

- [x] Contains schema changes.
- [ ] Schema changes require migration of existing data.
- [ ] Contains migration script for existing data.
- [ ] Changes should be covered by tests.
- [ ] Changes are covered by tests.

### Does this have a design impact ?
- [x] Yes 
- [ ] No

Zharktas commented on ckanext/ckanext-apicatalog_scheming/ckanext/apicatalog_scheming/i18n/ckanext-apicatalog_scheming.pot:
```diff
@@ -82,3 +82,64 @@ msgstr ""
+msgid ""
+"You can use <a href=\"#markdown\" title=\"Markdown quick "
+"reference\">Markdown formatting</a> here."
+msgstr ""
+
+#: ckanext/apicatalog_scheming/schemas/organization.json:21
+msgid "Description in Finnish"
+msgstr ""
+
+#: ckanext/apicatalog_scheming/schemas/organization.json:21
+msgid "Description in Swedish"
+msgstr ""
+
+#: ckanext/apicatalog_scheming/schemas/organization.json:21
+msgid "Description in English"
+msgstr ""
+
+#: ckanext/apicatalog_scheming/schemas/organization.json:36
+msgid "Other information and settings"
+msgstr ""
+
+#: ckanext/apicatalog_scheming/schemas/organization.json:35
+msgid "Organisation logo"
+msgstr ""
+
+#: ckanext/apicatalog_scheming/schemas/organization.json:37
+msgid "The logo is displayed on the organization page in API Catalogue."
+msgstr ""
+
+#: ckanext/apicatalog_scheming/schemas/organization.json:56
+msgid "If possible, provide a telephone number for the organization's switchboard or customer service, not an individual number."
+msgstr ""
+
+#: ckanext/apicatalog_scheming/schemas/organization.json:64
+msgid "If possible, enter the email address of the organization or customer service, not the email address of the individual."
+msgstr ""
+
+#: ckanext/apicatalog_scheming/templates/scheming/organization/group_form.html:28
+msgid "Discard changes"
+msgstr ""
```
This actually doesn't work. When extract_messages is run, these will be erased as they aren't marked as translatable strings in the code base. Translated strings in schema files need to be added to https://github.com/vrk-kpa/api-catalog/blob/master/ckanext/ckanext-apicatalog_scheming/ckanext/apicatalog_scheming/translations.py for translation engine to pick them up.

bertvi replied:
Added translations into translation.py file

Zharktas reviewed (changes requested):
Styling does not match the design, it needs to be updated as well. Translations are not generated from the codebase so they actually do not work.

bertvi commented:
> Styling does not match the design, it needs to be updated as well. Translations are not generated from the codebase so they actually do not work.

Updated the ui according to the design

Zharktas commented:
Now this doesn't work at all and gives an server error then trying to create new organization.  You should also generate the translations to match them with whatever the codebase has.

bertvi commented:
> Now this doesn't work at all and gives an server error then trying to create new organization. You should also generate the translations to match them with whatever the codebase has.

Now the "new organization" form works again. Translations still do not work, but I dont understand why, I have added translations to .pot file and also translations.py file, did the "tx push -s", translated these in Transfiex, performed "tx pull -f" and compiled the python code.

Zharktas commented on ckanext/ckanext-apicatalog_scheming/ckanext/apicatalog_scheming/schemas/organization.json:
```diff
@@ -5,27 +5,27 @@
+      "show_url": true,
       "display_snippet": null
     },
-    {
-      "field_name": "name",
-      "label": "URL",
-      "validators": "not_empty unicode name_validator group_name_validator",
-      "form_snippet": "slug.html",
-      "form_placeholder": "my-theme"
-    },
```
Name is mandatory field in CKAN, it cannot be removed, otherwise you can't create new organizations.

bertvi replied:
Added that field back to the schema and fixed the designs of new and edit forms to match

Zharktas commented on ckanext/ckanext-apicatalog_scheming/ckanext/apicatalog_scheming/schemas/organization.json:
```diff
@@ -5,27 +5,27 @@
+      "label": "Organisation description",
+      "extra_label": "A general, concise, and easy-to-understand description of the organization",
+      "input_title": ["Description in Finnish", "Description in Swedish", "Description in English"],
       "validators": "fluent_text mark_as_modified_in_catalog_if_changed",
```
Only default language should be required, now all the languages are required.

bertvi replied:
According to the design in Abstract, the description fields are all marked as required for all languages.

Zharktas replied:
You are actually correct, I accidentally looked at the wrong field. :D

Zharktas reviewed (changes requested):
New organizations still can't be created as the validators fail.

And the styling is still a way off from the design.

Zharktas commented on ckanext/ckanext-apicatalog_scheming/ckanext/apicatalog_scheming/schemas/organization.json:
```diff
@@ -5,27 +5,34 @@
       "validators": "fluent_text mark_as_modified_in_catalog_if_changed",
-      "form_placeholder": "A little information about my organization...",
-      "form_languages": ["fi", "sv", "en"]
+      "required": true,
+      "only_default_lang_required": true,
```
only_default_lang_required attribute only affects if the related validator is used. required attribute still makes everything mandatory. However as I said in flowdock, we need to leave this completely optional as descriptions do not come with ListMembers.

bertvi replied:
Done

Zharktas reviewed (changes requested):
Name field is still missing from the schema, so no new organizations can be created. The styling is closer to the design, but there is still work to do:
* The header should be outside of the white box.
* borders of the input fields are too thick.
* There should be morre margins between the input fields.
* The divider line between the fields should be the same size as the fields.
* The last 4 input fields are not a full width.

bertvi commented:
> Name field is still missing from the schema, so no new organizations can be created. 

organzation.json lines 18-24:

    {
      "field_name": "name",
      "label": "URL",
      "validators": "not_empty unicode name_validator group_name_validator",
      "form_snippet": null,
      "form_placeholder": "my-theme"
    },

Zharktas commented:
> > Name field is still missing from the schema, so no new organizations can be created.
> 
> organzation.json lines 18-24:
> 
> ```
> {
>   "field_name": "name",
>   "label": "URL",
>   "validators": "not_empty unicode name_validator group_name_validator",
>   "form_snippet": null,
>   "form_placeholder": "my-theme"
> },
> ```

When form_snippet is null, it does not exist in the form. Hence the validation will fail.

bertvi commented:
> Name field is still missing from the schema, so no new organizations can be created. The styling is closer to the design, but there is still work to do:
> 
> * The header should be outside of the white box.
> * borders of the input fields are too thick.
> * There should be morre margins between the input fields.
> * The divider line between the fields should be the same size as the fields.
> * The last 4 input fields are not a full width.

Updated

Zharktas reviewed (changes requested):
Styles should be in less files, so no inline styles. Margins should also come from styles and not extra semantic elements.

bertvi commented:
> Styles should be in less files, so no inline styles. Margins should also come from styles and not extra semantic elements.

Styles removed and placed in less files. Margins also come from styles.

Zharktas reviewed (changes requested):
The design seems okay now, but there are still some issues.

* Existing data is not used in the form when editing organization.
* When creating new organization the action buttons are update and cancel, there should be a save button instead.
* When modifying organization, the breadcrumb says add an organization.
* The logo buttons don't do anything.

bertvi commented:
> The design seems okay now, but there are still some issues.
> 
> * Existing data is not used in the form when editing organization.
> * When creating new organization the action buttons are update and cancel, there should be a save button instead.
> * When modifying organization, the breadcrumb says add an organization.
> * The logo buttons don't do anything.
> * The name is still missing from the form when creating organization so no new organizations can be created for testing purposes.

The logo buttons work again and the name field is back.

Zharktas commented:
This seems almost done, the input elements have quite a lot of margins between them which isn't in the design and organization logo has image_url text when url or file has been uploaded.

bertvi commented:
> This seems almost done, the input elements have quite a lot of margins between them which isn't in the design and organization logo has image_url text when url or file has been uploaded.

I added url_label, which was missing and removed margins from input fields

Zharktas commented on ckanext/ckanext-apicatalog_scheming/ckanext/apicatalog_scheming/templates/scheming/organization/group_form.html:
```diff
@@ -21,16 +23,17 @@
-    <div class="form-actions">
-        {% block delete_button %}
-        {% if action == 'edit' %}
-          {% if h.check_access('group_delete', {'id': data.id}) and action=='edit'  %}
-            {% set locale = h.dump_json({'content': _('Are you sure you want to delete this Organization?')}) %}
-            <a class="btn btn-danger pull-left" href="{% url_for controller='organization', action='delete', id=data.id %}" data-module="confirm-action" data-module-i18n="{{ locale }}">{% block delete_button_text %}{{ _('Delete') }}{% endblock %}</a>
-          {% endif %}
```
This check should exist, sysadmins must have a way to delete the organization. However it should check organization_delete instead.

bertvi replied:
I have now put that back

Zharktas reviewed (changes requested):
One minor fix on the delete button and this is ok.

Zharktas commented:
Blocks can be only defined once, so now this fails in a error.

Zharktas commented on ckanext/ckanext-apicatalog_scheming/ckanext/apicatalog_scheming/templates/scheming/organization/group_form.html:
```diff
@@ -23,8 +23,29 @@
+    <div class="form-actions">
+        {% block delete_button %}
+        {% if action == 'edit' %}
+          {% if h.check_access('group_delete', {'id': data.id}) and action=='edit'  %}
```
This should be `h.check_access('organization_delete'`

bertvi replied:
Updated

bertvi commented on ckanext/ckanext-apicatalog_scheming/ckanext/apicatalog_scheming/templates/scheming/organization/group_form.html:
```diff
@@ -23,8 +23,29 @@
+                        <a class="btn btn-danger pull-left" href="{% url_for controller='organization', action='delete', id=data.id %}" data-module="confirm-action" data-module-i18n="{{ locale }}">{% block delete_button_text %}{{ _('Delete') }}{% endblock %}</a>
+                    {% endif %}
+                {% endif %}
+            {% endblock %}
```
Removed the delete_button block and kept only the if clause
